nifty/pipeline/nifsUtils.py

```diff
@@ -1232,6 +1232,19 @@ def getFile(url):
         for chunk in r.iter_content(chunk_size=128):
             f.write(chunk)
 
+    # Do MD5 Verification of the file; raise IO error if a problem happened.
+    try:
+        server_checksum = r.headers['Content-MD5']
+        with open(filename, 'rb') as f:
+            download_checksum = hashlib.md5(f.read()).hexdigest()
```

Comment:
GEMINI files are not that big but in general this could be an expensive (e.g. time consuming) operation. Best would be to feed the bytes to `hashlib` at the same time you are writing them into the file (line 1233). That's if the `Content-MD5` present.

Reply:
Awesome, good idea. Added in b88d045.